pkg/mco/machineconfig.go

```diff
@@ -0,0 +1,283 @@
+
+// Update renovates the existing machineconfig object with machineconfig definition in builder.
+func (builder *MCBuilder) Update() (*MCBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating machineconfig %s", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.MachineConfigs().Update(
+		context.TODO(), builder.Definition, metav1.UpdateOptions{})
+
+	return builder, err
+}
+
+// Exists checks whether the given machineconfig exists.
+func (builder *MCBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if the MachineConfig object %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.MachineConfigs().Get(
+		context.Background(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithLabel redefines machineconfig definition with the given label.
+func (builder *MCBuilder) WithLabel(key string, value string) *MCBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Labeling the machineconfig %s with %s=%s", builder.Definition.Name, key, value)
+
+	if key == "" {
+		glog.V(100).Infof("The key can't be empty")
+
+		builder.errorMsg = "'key' cannot be empty"
+
+		return builder
+	}
+
+	if builder.Definition.Labels == nil {
+		builder.Definition.Labels = map[string]string{}
+	}
+
+	builder.Definition.Labels[key] = value
+
+	return builder
+}
+
+// WithOptions creates the machineconfig with generic mutation options.
+func (builder *MCBuilder) WithOptions(options ...MCAdditionalOptions) *MCBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting machineconfig additional options")
+
+	for _, option := range options {
+		if option != nil {
+			builder, err := option(builder)
+
+			if err != nil {
+				glog.V(100).Infof("Error occurred in mutation function")
+
+				builder.errorMsg = err.Error()
+
+				return builder
+			}
+		}
+	}
+
+	return builder
+}
+
+// WithKernelArguments sets the specified KernelArguments to the MachineConfig.
+func (builder *MCBuilder) WithKernelArguments(kernelArgs []string) *MCBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting KernelArguments: %v", kernelArgs)
+
+	builder.Definition.Spec.KernelArguments = kernelArgs
+
+	return builder
+}
+
+// WithExtensions sets the specified Extensions to the MachineConfig.
+func (builder *MCBuilder) WithExtensions(extensions []string) *MCBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting Extensions: %v", extensions)
+
+	builder.Definition.Spec.Extensions = extensions
+
+	return builder
+}
+
+// WithFips sets the specified FIPS value to the MachineConfig.
+func (builder *MCBuilder) WithFips(fips bool) *MCBuilder {
```

Comment:
nit: WithFIPS

Reply:
Update it to be coherent with the K8s definition.

Thanks!